File: evidence/generated-systems/angular/governance-on-angular/src/app/models/RecordsRepository.js

// Define collection and schema for RecordsRepository
export interface RecordsRepository {
    name:
	type : string
    location:
	type : string
    ownerDepartment:
	type : string
    Organization:
	type : Schema.Types.ObjectId
    Records:
 	type : [{ type: Schema.Types.ObjectId, ref: 'Record_' }]
    Systems:
 	type : [{ type: Schema.Types.ObjectId, ref: 'System_' }]
    RetentionSchedules:
 	type : [{ type: Schema.Types.ObjectId, ref: 'RetentionSchedule' }]
    LegalHolds:
 	type : [{ type: Schema.Types.ObjectId, ref: 'LegalHold' }]
    RepositoryType:
 	type : String
#
    collection: 'recordsRepositorys'
}

File: evidence/generated-systems/golang/governance-on-golang/governance-on-golang/internal/model/RecordsRepository.go
package model

import (
    "gorm.io/gorm"
)

//==============================================================
// RecordsRepository Declaration
//==============================================================
type RecordsRepository struct {
    gorm.Model
     Name                                    string
    Location                                    string
    OwnerDepartment                                    string
    OrganizationId         *uint
    Organization           *Organization `gorm:"foreignKey:OrganizationId"`
     Records           []Record_ `gorm:"foreignKey:RecordsFromRecordsRepositoryId"`
     Systems           []System_ `gorm:"foreignKey:SystemsFromRecordsRepositoryId"`
     RetentionSchedules           []RetentionSchedule `gorm:"foreignKey:RetentionSchedulesFromRecordsRepositoryId"`
     LegalHolds           []LegalHold `gorm:"foreignKey:LegalHoldsFromRecordsRepositoryId"`
    RepositoryType                      RepositoryType

// parent associations as their child

}

